webui.py

```diff
@@ -884,22 +884,24 @@ def dump_default_english_config():
 
 # dump_default_english_config()
 
-app, _, _ = shared.gradio_root.launch(
-    inbrowser=False,
-    server_name=args_manager.args.listen,
-    server_port=args_manager.args.port,
-    share=args_manager.args.share,
-    auth=check_auth if args_manager.args.share and auth_enabled else None,
-    blocked_paths=[constants.AUTH_FILENAME],
-    prevent_thread_lock=True,
-    app_kwargs={
-        "docs_url": "/docs",
-        "redoc_url": "/redoc",
-    },
-)
 
-app = create_api(app, generate_clicked, refresh_seed, recover_task, stop_clicked, skip_clicked, trigger_describe)
-script_callbacks.app_started_callback(None, app)
+def make_app(server_port, refresh_seed, trigger_describe):
+    app, _, _ = shared.gradio_root.launch(
+        inbrowser=False,
+        server_name=args_manager.args.listen,
+        server_port=server_port,
+        share=args_manager.args.share,
+        auth=check_auth if args_manager.args.share and auth_enabled else None,
+        blocked_paths=[constants.AUTH_FILENAME],
+        prevent_thread_lock=True,
+        app_kwargs={
+            "docs_url": "/docs",
+            "redoc_url": "/redoc",
+        },
+    )
+
+    app = create_api(app, generate_clicked, refresh_seed, recover_task, stop_clicked, skip_clicked, trigger_describe)
```

Comment:
I would suggest to separate ui creation and api creation, since api doesn't depend on gradio ui.

I would suggest to change `def  make_ui():` to `def  make_gradio_ui():`

I would suggest to change `def make_app(server_port, refresh_seed, trigger_describe):` to `def mount_gradio_app(server_port, refresh_seed, trigger_describe):`, where

```python
def mount_gradio_app(gradio_root, server_port, refresh_seed, trigger_describe):
    with gradio_root:
        make_ui(refresh_seed, trigger_describe)
        app, _, _ = gradio_root.launch(
            inbrowser=False,
            server_name=args_manager.args.listen,
            server_port=server_port,
            share=args_manager.args.share,
            auth=check_auth if args_manager.args.share and auth_enabled else None,
            blocked_paths=[constants.AUTH_FILENAME],
            prevent_thread_lock=True,
            app_kwargs={
                "docs_url": "/docs",
                "redoc_url": "/redoc",
            },
        )
```

And then change the start function 
```python
def start(server_port: int = 0):
    if server_port == 0:
        server_port = args_manager.args.port
    script_callbacks.before_ui_callback()
    app = mount_gradio_app(shared.gradio_root, server_port, refresh_seed, trigger_describe)
    app = create_api(app, generate_clicked, refresh_seed, recover_task, stop_clicked, skip_clicked, trigger_describe)
    worker.start(app)
    asyncio.run(block_thread())
```